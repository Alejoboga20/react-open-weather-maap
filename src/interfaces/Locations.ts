export interface Locations {
  label: string;
  value: string;
  location: Coordinates;
}

export interface Coordinates {
  latitude: number;
  longitude: number;
}
