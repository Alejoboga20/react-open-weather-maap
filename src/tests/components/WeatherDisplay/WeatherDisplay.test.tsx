import { shallow } from 'enzyme';
import {
  WeatherDisplay,
  WeatherDisplayProps,
} from '../../../components/WeatherDisplay/WeatherDisplay';
import { Description, Main } from '../../../interfaces/WeatherResponse';

const weatherDisplayProps: WeatherDisplayProps = {
  dailyTemp: [
    {
      dt: 10,
      sunrise: 10,
      sunset: 10,
      moonrise: 10,
      moonset: 10,
      moon_phase: 10,
      temp: {
        day: 10,
        min: 10,
        max: 10,
        night: 10,
        eve: 10,
        morn: 10,
      },
      feels_like: {
        day: 10,
        night: 10,
        eve: 10,
        morn: 10,
      },
      pressure: 10,
      humidity: 10,
      dew_point: 10,
      wind_speed: 10,
      wind_deg: 10,
      wind_gust: 10,
      weather: [
        {
          id: 10,
          main: Main.Clear,
          description: Description.BrokenClouds,
          icon: 'Cloud',
        },
      ],
      clouds: 10,
      pop: 10,
      uvi: 10,
    },
  ],
};

describe('WeatherDisplay Tests', () => {
  const wrapper = shallow(<WeatherDisplay {...weatherDisplayProps} />);

  test('should Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should Label and Item', () => {
    expect(wrapper.find('Label').exists()).toBe(true);
    expect(wrapper.find('WeatherItem').exists()).toBe(true);
  });

  test('should not display Label not Items', () => {
    const wrapper = shallow(<WeatherDisplay />);
    expect(wrapper.find('Label').exists()).toBe(false);
    expect(wrapper.find('WeatherItem').exists()).toBe(false);
  });
});
