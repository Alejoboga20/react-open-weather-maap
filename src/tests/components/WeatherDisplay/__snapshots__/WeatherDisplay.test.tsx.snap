// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDisplay Tests should Match Snapshot 1`] = `
<div
  className="weatherDisplay_container"
>
  <Label
    labelContent="Temperature for the following days is: "
  />
  <WeatherItem
    key="10"
    temp={10}
  />
</div>
`;
